misc/json-to-record-converter/src/main/java/io/ballerina/jsonmapper/JsonToRecordMapper.java

```diff
@@ -633,6 +630,25 @@ private static TypeDescriptorNode createUnionTypeDescriptorNode(List<TypeDescrip
                 NodeFactory.createOptionalTypeDescriptorNode(parenTypeDescNode, questionMarkToken) : parenTypeDescNode;
     }
 
+    private static TypeDescriptorNode joinToUnionTypeDescriptorNode(List<TypeDescriptorNode> typeNames) {
+        Token pipeToken = NodeFactory.createToken(SyntaxKind.PIPE_TOKEN);
+
+        if (typeNames.size() == 0) {
+            Token typeName = AbstractNodeFactory.createToken(SyntaxKind.ANYDATA_KEYWORD);
+            return NodeFactory.createBuiltinSimpleNameReferenceNode(typeName.kind(), typeName);
+        } else if (typeNames.size() == 1) {
+            return typeNames.get(0);
```

Comment:
Repeated logic. Already there in https://github.com/ballerina-platform/ballerina-lang/pull/41091/files#diff-04c68038ff3884caedd75f69d092557245fce262b64ad5de444d82ecb77bdc58L615-L619